test/js/store/projects/actions.test.js

```diff
@@ -10,6 +12,49 @@ describe('updateProject', () => {
   });
 });
 
+describe('setUsersOnProject', () => {
+  test('PUTs to API', async () => {
+    const store = storeWithThunk({});
+    const project = {
+      id: 'project-id',
+      name: 'My Project',
+      github_users: [
+        {
+          id: '123456',
+          login: 'TestGitHubUser',
+          avatar_url: 'https://example.com/avatar.png',
+        },
+      ],
+    };
+    const action = {
+      type: 'PROJECT_UPDATE',
+      payload: project,
+    };
+    fetchMock.putOnce(window.api_urls.project_detail(project.id), project);
+    await store.dispatch(actions.setUsersOnProject(project));
+    const allActions = store.getActions();
+
+    expect(allActions[0]).toEqual(action);
+  });
+  test('logs an error if an error happens', async () => {
+    const store = storeWithThunk({});
+    const project = {
+      id: 'project-id',
+      name: 'My Project',
+      github_users: [
+        {
+          id: '123456',
+          login: 'TestGitHubUser',
+          avatar_url: 'https://example.com/avatar.png',
+        },
+      ],
+    };
+    fetchMock.putOnce(window.api_urls.project_detail(project.id), 500);
+    await store.dispatch(actions.setUsersOnProject(project));
+    return expect(window.console.error).toHaveBeenCalled();
```

Comment:
@wlonk it's not obvious to me why this test isn't hitting the try block to throw an error. I am curious, should this also dispatch a failed action, like the the other actions in this file? Would that make testing this easier?

Reply:
I'm not sure either; I'm experimenting with it right now, and that's a good way to go probably.